lib/ansible/modules/packaging/os/pacman.py

```diff
@@ -138,21 +138,62 @@ def get_version(pacman_output):
     return None
 
 
+def find_provider(module, pacman_path, target_package):
+    """Find the package providing `target_package`"""
+    cmd = "%s -Q %s" % (pacman_path, target_package)
+    rc, output, stderr = module.run_command(cmd, check_rc=False)
+    if rc == 0:
+        realname = output.split()[0]
+        version = output.split()[1]
+        return (realname, version)
+
+    # Unable to find package named `target_package` - scan full package dump
+    cmd = "%s -Qqei" % pacman_path
+    rc, output, stderr = module.run_command(cmd, check_rc=False)
+    if rc != 0:
+        # some unknown error
+        return (None, None)
+
+    target_package = target_package.lower()
+    provider = None
+    version = None
+
+    packages = output.split('\n\n')
+
+    for package in packages:
+        lines = package.split('\n')
+        pkg_name = pkg_version = None
+        pkg_provides = []
+
+        for line in lines:
+            if line.startswith("Name"):
+                pkg_name = line.split(':', 1)[1].strip().lower()
```

Comment:
```suggestion
                pkg_name = line.partition(':')[-1].strip()
```
idem, no need to lower.

Reply:
If you're going to do partition, I'd rather use the fact that it'll always return a 3-tuple instead of using `-1` to index the third element. i.e.:

```python
line.partition(':')[2]
```

Reply:
@thomwiggers `line.partition(':')[2]` yes, I have no problem with that.